--- src/dtlock/LockDialog.cpp.orig	2011-07-09 10:44:46.000000000 +0300
+++ src/dtlock/LockDialog.cpp	2012-10-10 16:15:54.000000000 +0300
@@ -1,6 +1,8 @@
 #include "LockDialog.h"
 
 #include <pwd.h>
+#include <unistd.h>
+#include <crypt.h>
 
 #ifdef PLATFORM_LINUX
   #include <shadow.h>
--- src/dtlogin/LoginWindow.cpp.orig	2011-07-02 15:09:23.000000000 +0300
+++ src/dtlogin/LoginWindow.cpp	2012-10-10 16:18:18.000000000 +0300
@@ -1,6 +1,7 @@
 #include "LoginWindow.h"
 
 #include <pwd.h>
+#include <unistd.h>
 
 #ifdef PLATFORM_LINUX
   #include <shadow.h>
--- src/dtlogin/main.cpp.orig	2011-06-26 22:41:07.000000000 +0300
+++ src/dtlogin/main.cpp	2012-10-10 16:17:32.000000000 +0300
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include <motifmm.h>
 #include <opencde.h>
 
